azure-quantum/azure/quantum/qiskit/provider.py

```diff
@@ -120,18 +120,37 @@ def backends(self, name=None, **kwargs):
             backend_list, filters=workspace_allowed, **kwargs
         )
 
-        # Also filter out non-default backends.
-        default_backends = list(
-            filter(
-                lambda backend: self._match_all(
-                    backend.configuration().to_dict(), {"is_default": True}
-                ),
-                filtered_backends,
-            )
-        ) 
-       # If default backends were found - return them, otherwise return the filtered_backends collection.
-       # The latter case could happen where there's no default backend defined for the specified target.  
-        if len(default_backends) > 0:
+        # The `is_passthrough` flag is used as a more intuitive way to match passthrough backends, and
+        # to allow for us to set default passhtrough backends, while still defaulting to QIR backends
+        # in case we match to both a QIR and a passthrough backend. Without the `is_passthrough`flag 
+        # fetching backends for IonQ require at least 2 input flags to be able to match to a single
+        # backend, those being both `input_data_format` and `gateset` (as there are multiple gateset
+        # versions per passthrough backend) 
+        def filter_non_passthrough_defaults(filtered_backends):
```

Comment:
I think I haven't understood the purpose of this function. What made the `_filter_backends` method no longer sufficient to get the filtered list of backends we now expect?

Reply:
`_filter_backends`'s default backend filtration logic previously only considered `is_default` flag... now, given we have both passthrough and QIR defaults, we must consider the `is_passthrough` flag, and if there is no `is_passthrough` flag, we prefer QIR defaults over passthrough defaults. So, the default selection / filtration logic got a bit more complicated

Reply:
After thinking about it, I believe having extra logic to being able to create a pass-through backend is not worth it. The reason is that functionally, a pass-through backend does not offer additional features than a QIR backend, with the exception of IonQ's native backends that provide a specialized gateset (in which case a user would filter based on the `gateset` configuration property).

We should consider using pass-through backends an advanced feature, so it's ok to have to specify multiple filtering properties is needed (like in the case of IonQ).